server/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -177,6 +187,8 @@ private synchronized void closeWithTragicEvent(final Exception ex) {
         minSeqNo = SequenceNumbers.min(minSeqNo, seqNo);
         maxSeqNo = SequenceNumbers.max(maxSeqNo, seqNo);
 
+        nonFsyncedSequenceNumbers.add(seqNo);
```

Comment:
I have increased the initial capacity from the default (4) to 64. Even though the resizing will be done under the block, I'm not too worried about performance here as it's usually a simple array copy, and typically a pretty small one at that (bulk requests have typically up to a few thousand items, and they are fsynced on a per-request level). LocalCheckPointTracker also allocates that kind of memory under lock, and we haven't seen contention on that one. We can look at the nightly benchmarks to see whether performance is negatively impacted. It would probably take a very beefy machine (lots of threads) for this to become apparent.

Reply:
Sounds good.